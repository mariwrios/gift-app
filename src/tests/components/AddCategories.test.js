import React from "react"
import "@testing-library/jest-dom"
import { shallow } from "enzyme"
import AddCategories from "../../components/AddCategories"

describe("Probando el componente AddCategories", () => {
  const setCategories = jest.fn()
  let wrapper = shallow(<AddCategories setCategories={setCategories} />)

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategories setCategories={setCategories} />)
  })
  test("Debe mostrarse correctamente <AddCategories/>", () => {
    expect(wrapper).toMatchSnapshot()
  })

  test("Debe cambiarse el contenido del input correctamente", () => {
    const input = wrapper.find("input")
    const value = "Hola Mundo"
    input.simulate("change", { target: { value } })
  })

  test("No debe activarse el onSumbit si no se le envia nada", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} })

    expect(setCategories).not.toHaveBeenCalled()
  })
  test("Debe llamar el setCategories y limpiar la caja de texto", () => {
    const value = "Naruto"
    const inputValue = wrapper.find("input").prop("value")

    wrapper.find("input").simulate("change", { target: { value } })

    wrapper.find("form").simulate("submit", { preventDefault() {} })

    expect(setCategories).toHaveBeenCalledWith(expect.any(Function))
    //aqui verificamos que se llamara la menos una vez y que recibio una funion
    expect(inputValue).toBe("")
  })
})
