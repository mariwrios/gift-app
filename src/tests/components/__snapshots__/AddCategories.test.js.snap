// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente AddCategories Debe mostrarse correctamente <AddCategories/> 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Buscar..."
    type="text"
    value=""
  />
</form>
`;
